library/Icingadb/View/ServicegroupGridRenderer.php

```diff
@@ -145,60 +160,109 @@ protected function createGroupBadge(): Link
+                        $item->services_unknown_handled,
+                        $item->display_name
                     )
                 ]
             );
-        } elseif ($this->item->services_pending > 0) {
+        } elseif ($item->services_pending > 0) {
             $urlFilter->add(Filter::equal('service.state.soft_state', 99));
 
-            return new Link(
-                new StateBadge($this->item->services_pending, 'pending'),
+            $link = new Link(
+                new StateBadge($item->services_pending, 'pending'),
                 $url->setFilter($urlFilter),
                 [
                     'title' => sprintf(
                         $this->translatePlural(
                             'List %d service that is currently in PENDING state in service group "%s"',
                             'List %d services which are currently in PENDING state in service group "%s"',
-                            $this->item->services_pending
+                            $item->services_pending
                         ),
-                        $this->item->services_pending,
-                        $this->item->display_name
+                        $item->services_pending,
+                        $item->display_name
                     )
                 ]
             );
-        } elseif ($this->item->services_ok > 0) {
+        } elseif ($item->services_ok > 0) {
             $urlFilter->add(Filter::equal('service.state.soft_state', 0));
 
-            return new Link(
-                new StateBadge($this->item->services_ok, 'ok'),
+            $link = new Link(
+                new StateBadge($item->services_ok, 'ok'),
                 $url->setFilter($urlFilter),
                 [
                     'title' => sprintf(
                         $this->translatePlural(
                             'List %d service that is currently in OK state in service group "%s"',
                             'List %d services which are currently in OK state in service group "%s"',
-                            $this->item->services_ok
+                            $item->services_ok
                         ),
-                        $this->item->services_ok,
-                        $this->item->display_name
+                        $item->services_ok,
+                        $item->display_name
+                    )
+                ]
+            );
+        } else {
+            $link = new Link(
+                new StateBadge(0, 'none'),
+                $url,
+                [
+                    'title' => sprintf(
+                        $this->translate('There are no services in service group "%s"'),
+                        $item->display_name
+                    )
+                ]
+            );
+        }
+
+        $visual->addHtml($link);
+    }
+
+    public function assembleTitle($item, HtmlDocument $title, string $layout): void
+    {
+        if ($layout !== 'header') {
```

Comment:
That renderer isn't used for a header, right?

Reply:
You are right.